### PR TITLE
Update build submodule and use helm v2.17.0

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Helm recently changed the location of stable charts, which is
automatically handled in the latest patch release. This updates the
build submodule to pick up the latest patch and get our CI pipelines
working again.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

`make build`

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by jbw976 — COMMENTED** [review]
a few of the CI checks failed, and they look related to the helm stable charts issue.  do we expect these to start passing again AFTER this is merged?  I know how Jenkins used to behave during CI builds for PRs that change build files, but I'm not sure how GH Actions behaves.

**Comment by hasheddan** [comment]
@jbw976 I believe it is because we are still falling back to helm 2 here, which was not updated. I am going to bump to helm 3.

**Comment by hasheddan** [comment]
Note that we will be blocked on update to helm 3 until https://github.com/helm/helm/pull/9118 is merged.

**Comment by hasheddan** [comment]
@jbw976 this should be good to go now :+1:

File: Makefile
Comment (jbw976) [inline Makefile:41]: do we still need to update the build submodule if we are using this newer version of helm2?

File: cluster/local/integration_tests.sh
Comment (jbw976) [inline cluster/local/integration_tests.sh:46]: what caused this casing to need to change? something in the build submodule set of updates?

File: Makefile
Comment (hasheddan) [reply Makefile:41]: We don't, but I think it is good to continue to update frequently. In reality I think we should have the build submodule versioned, but right now the latest commit it essentially the latest version.

File: cluster/local/integration_tests.sh
Comment (hasheddan) [reply cluster/local/integration_tests.sh:46]: kind version bump in build submodule, we saw the same change required in other repos 👍

**Review by jbw976 — APPROVED** [review]

**Pull request merged**
